test/Datadog.Trace.ClrProfiler.IntegrationTests/AspNet/AspNetWebApi2Tests.cs

```diff
@@ -46,58 +66,94 @@ public AspNetWebApi2TestsCallTargetClassic(IisFixture iisFixture, ITestOutputHel
     public class AspNetWebApi2TestsCallTargetIntegrated : AspNetWebApi2Tests
     {
         public AspNetWebApi2TestsCallTargetIntegrated(IisFixture iisFixture, ITestOutputHelper output)
-            : base(iisFixture, output, enableCallTarget: true, classicMode: false)
+            : base(iisFixture, output, enableCallTarget: true, classicMode: false, enableFeatureFlag: false)
+        {
+        }
+    }
+
+    [Collection("IisTests")]
+    public class AspNetWebApi2TestsCallTargetClassicWithFeatureFlag : AspNetWebApi2Tests
+    {
+        public AspNetWebApi2TestsCallTargetClassicWithFeatureFlag(IisFixture iisFixture, ITestOutputHelper output)
+            : base(iisFixture, output, enableCallTarget: true, classicMode: true, enableFeatureFlag: true)
         {
         }
     }
 
+    [Collection("IisTests")]
+    public class AspNetWebApi2TestsCallTargetIntegratedWithFeatureFlag : AspNetWebApi2Tests
+    {
+        public AspNetWebApi2TestsCallTargetIntegratedWithFeatureFlag(IisFixture iisFixture, ITestOutputHelper output)
+            : base(iisFixture, output, enableCallTarget: true, classicMode: false, enableFeatureFlag: true)
+        {
+        }
+    }
+
+    [UsesVerify]
     public abstract class AspNetWebApi2Tests : TestHelper, IClassFixture<IisFixture>
     {
         private readonly IisFixture _iisFixture;
+        private readonly string _testName;
 
-        public AspNetWebApi2Tests(IisFixture iisFixture, ITestOutputHelper output, bool enableCallTarget, bool classicMode)
+        public AspNetWebApi2Tests(IisFixture iisFixture, ITestOutputHelper output, bool enableCallTarget, bool classicMode, bool enableFeatureFlag)
             : base("AspNetMvc5", @"test\test-applications\aspnet", output)
         {
             SetServiceVersion("1.0.0");
             SetCallTargetSettings(enableCallTarget);
+            if (enableFeatureFlag)
+            {
+                SetEnvironmentVariable(ConfigurationKeys.FeatureFlags.RouteTemplateResourceNamesEnabled, "true");
+            }
 
             _iisFixture = iisFixture;
             _iisFixture.TryStartIis(this, classicMode);
+            _testName = nameof(AspNetWebApi2Tests)
+                      + (enableCallTarget ? ".CallSite" : ".CallTarget")
+                      + (classicMode ? ".Classic" : ".Integrated")
+                      + (enableFeatureFlag ? ".NoFF" : ".WithFF")
+                      + (RuntimeInformation.ProcessArchitecture == Architecture.X64 ? ".X64" : ".X86"); // assume that arm is the same
         }
 
+        public static TheoryData<string, int> Data() => new()
```

Comment:
This is so nice, we just need to specify the inputs and ensure the output is expected 😍